drivers/soundwire/mipi_disco.c

```diff
@@ -284,6 +284,34 @@ static int sdw_slave_read_dpn(struct sdw_slave *slave,
 	return 0;
 }
 
+/*
+ * In MIPI DisCo spec for SoundWire, lane mapping for a slave device is done with
+ * mipi-sdw-lane-x-mapping properties, where x is 1..7, and the values for those
+ * properties are mipi-sdw-manager-lane-x or mipi-sdw-peripheral-link-y, where x
+ * is an integer between 1 to 7 if the lane is connected to a manager lane, y is a
+ * character between A to E if the lane is connected to another peripheral lane.
+ */
+static int sdw_slave_read_lane_mapping(struct sdw_slave *slave)
+{
+	struct sdw_slave_prop *prop = &slave->prop;
+	struct device *dev = &slave->dev;
+	char prop_name[30];
+	const char *prop_val;
+	size_t len;
+	int ret, i;
+
+	for (i = 0; i < SDW_MAX_LANES; i++) {
+		snprintf(prop_name, sizeof(prop_name), "mipi-sdw-lane-%d-mapping", i);
+		ret = device_property_read_string(dev, prop_name, &prop_val);
+		if (ret)
+			continue;
+		len = strlen(prop_val);
+		/* The last character is enough to identify the connection */
+		prop->lane_maps[i] = prop_val[len - 1];
```

Comment:
Should this be `prop->lane_maps[i] = prop_val[len - 1] - '0';`? `prop_val[len - 1]` is a char and the value is '1' to '7'. And we treat prop->lane_maps[] as integer, right?

Reply:
I don't know, I took Chao's code as is...

Reply:
> I don't know, I took Chao's code as is...

I would suggest that we treat prop->lane_maps[] as integer, and convert '1' to 1 here.

Reply:
> > Should this be `prop->lane_maps[i] = prop_val[len - 1] - '0';`? `prop_val[len - 1]` is a char and the value is '1' to '7'. And we treat prop->lane_maps[] as integer, right?
> 
> Do we have `isodigit()`? Or `isdigit()` can be used.

@andy-shev Are you suggesting we check `isdigit(prop_val[len - 1])` before ` prop->lane_maps[i] = prop_val[len - 1] - '0';`?

Reply:
> > > Should this be `prop->lane_maps[i] = prop_val[len - 1] - '0';`? `prop_val[len - 1]` is a char and the value is '1' to '7'. And we treat prop->lane_maps[] as integer, right?
> > 
> > 
> > Do we have `isodigit()`? Or `isdigit()` can be used.
> 
> @andy-shev Are you suggesting we check `isdigit(prop_val[len - 1])` before ` prop->lane_maps[i] = prop_val[len - 1] - '0';`?

There are many ways of converting that. Once can use `is*()` API to check, or directly call something like `hex_to_bin()` first (from hex.h) and compare result to be in a range: `in_range(hex_to_bin(...), ...)`.
You may also leave the above, but it might be less illustrative on what's going on there.

Reply:
patch dropped, will the added by the multi-lane work in PR #4733